recipes/protobuf/all/test_package/conanfile.py

```diff
@@ -1,20 +1,32 @@
 from conans import ConanFile, CMake, tools
 import os
-
+from contextlib import contextmanager
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package_multi"
+    generators = "cmake", "cmake_find_package_multi", "VirtualRunEnv", "VirtualBuildEnv"
 
     def build_requirements(self):
-        if tools.cross_building(self.settings):
+        if hasattr(self, "settings_build"):
             self.build_requires(str(self.requires['protobuf']))
 
+    @contextmanager
+    def _env(self):
+        if hasattr(self, "settings_build"):
+            # 2 profiles - nothing is needed, VirtualRunEnv/VirtualBuildEnv manages the things
+            yield
+        else:
+            from conans import RunEnvironment
+            env_build = RunEnvironment(self)
+            with tools.environment_append(env_build.vars):
```

Comment:
```suggestion
            with tools.run_environment(self):
```

Reply:
Let's see if it works with 1 profile on Macos if shared, but I've tested with recent CMake versions when I've introduced the workaround for SIP.

Reply:
I need to test these changes locally, I am not sure if they are identical ATM